rotkehlchen/chain/ethereum/modules/lido/decoder.py

```diff
@@ -0,0 +1,126 @@
+            if (
+                event.address == self.steth_evm_address and
+                event.asset == A_ETH and
+                event.balance.amount == collateral_amount and
+                event.event_type == HistoryEventType.SPEND and
+                event.location_label == sender
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Submit {collateral_amount} ETH to Lido for receiving stETH in exchange'  # noqa: E501
+                event.counterparty = CPT_LIDO
+                #  preparing next action to be processed when erc20 transfer will be decoded by rotki  # noqa: E501
+                #  needed because submit event is emitted prior to the erc20 transfer
+                paired_event = event
+                action_from_event_type = HistoryEventType.RECEIVE
+                action_to_event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                action_to_notes = f'Receive {{amount}} stETH in exchange of the deposited ETH'  # {amount} to be set by transform ActionItem  # noqa: E501,F541
+                break
+        else:  # did not find anything
+            log.error(
+                f'At lido steth submit decoding of tx {context.transaction.tx_hash.hex()}'
+                f' did not find the related ETH transfer',
+            )
+            return DEFAULT_DECODING_OUTPUT
+
+        action_items = []  # also create an action item for the reception of the stETH tokens
+        if paired_event is not None and action_from_event_type is not None:
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=action_from_event_type,
+                from_event_subtype=HistoryEventSubType.NONE,
+                asset=A_STETH,
+                amount=collateral_amount,
+                amount_error_tolerance=FVal(token_raw_value_decimals(STETH_MAX_ROUND_ERROR_WEI, 18)),  # passing the maximum rounding error for finding the related stETH transfer  # noqa: E501
+                to_event_subtype=action_to_event_subtype,
+                to_notes=action_to_notes,
+                to_counterparty=CPT_LIDO,
+                paired_event_data=(paired_event, True),
+                extra_data={'staked_eth': str(collateral_amount)},
+            ))
+
+        return DecodingOutput(action_items=action_items, matched_counterparty=CPT_LIDO)
+
+    def _decode_lido_eth_staking_contract(self, context: DecoderContext) -> DecodingOutput:
+        """Decode interactions with stETH ans wstETH contracts"""
+
+        if (
+            context.tx_log.topics[0] == LIDO_STETH_SUBMITTED and
+            self.base.any_tracked([hex_or_bytes_to_address(context.tx_log.topics[1])])
+        ):
+            return self._decode_lido_staking_in_steth(context=context, sender=hex_or_bytes_to_address(context.tx_log.topics[1]))  # noqa: E501
```

Comment:
As I pointed out here: https://github.com/rotki/rotki/pull/7949#discussion_r1606113674

1. Use the walrus operator. This way you don't run `hex_or_bytes_to_address` twice.
2. Use is_tracked. It's just 1 address.

```suggestion
        if (
            context.tx_log.topics[0] == LIDO_STETH_SUBMITTED and
            self.base.is_tracked(sender := hex_or_bytes_to_address(context.tx_log.topics[1]))
        ):
            return self._decode_lido_staking_in_steth(context=context, sender=sender)
```